src/main/java/athena/task/Time.java

```diff
@@ -76,6 +74,11 @@ public Time(Boolean isFlexible, String startTime, String duration, String deadli
                 throw new TaskDuringSleepTimeException();
             }
         }
+        if (this.duration > 16) {
+            throw new TaskTooLongException(this.duration);
+        }
```

Comment:
Could you shift this into the isClashWithSleep method so it looks cleaner?

`if (duration > 16 && isFlexible) throw new TaskTooLongException(duration)`